apps/cowswap-frontend/src/modules/tokensList/hooks/useAllTokensBalances.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react'
+
+import { TokenWithLogo } from '@cowprotocol/common-const'
+import { useAllTokens } from '@cowprotocol/tokens'
+import { useWalletInfo } from '@cowprotocol/wallet'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+
+import { OnchainState, TokenAmounts, useOnchainBalances } from 'modules/tokens'
+
+import { useNativeBalance } from 'common/hooks/useNativeBalance'
+
+const defaultBalancesState: [TokenAmounts, boolean] = [{}, false]
+
+/**
+ * Returns balances for all tokens + native token
+ */
+export function useAllTokensBalances(): [TokenAmounts, boolean] {
+  const { account } = useWalletInfo()
+  const allTokens = useAllTokens()
+  const nativeBalance = useNativeBalance()
```

Comment:
Hmmm at first `useNativeBalance` being updated only once every minute wasn't such a big deal because I thought it would be used only in the header.
But if this is being used in more places like the token list, I argue it should be updated more often.

Reply:
Agree, changed to 30 seconds:
https://github.com/cowprotocol/cowswap/pull/3201/commits/19e7d6d800b066cfa59fbdc403a1784d8d1fcd43